ca/config.go

```diff
@@ -214,6 +189,11 @@ func (s *SecurityConfig) IssuerInfo() *IssuerInfo {
 	return s.issuerInfo
 }
 
+// Close closes the security config, freeing up the watch queue resource
+func (s *SecurityConfig) Close() {
```

Comment:
Not sure if it makes sense, but having `NewSecurityConfig` return a cancel function might make it harder to leak these. On the other hand, security configs might always be stored somewhere long-term, in which case it wouldn't help.

In general this change seems to be going in the right direction (I'm happy to see `SetWatch` go), but making the `SecurityConfig` something whose resources need to be consciously deallocated is unfortunate.